### ISSUE TITLE
Nittygriddy cannot download the dataset which I defined in nitty_datasets.yml

### ISSUE THREAD

**Issue by chhuangAli** [cbourjau/nittygriddy#27]
Hello,

 I am using the nittygriddy to download the raw data to test my task. But for the LHC16e2, I could not download it. I could find a relating solution on the Internet so I send this email to you. My laptop OS is  based on ubuntu16.04 and the AliPhysics with ROOT6. I defined the LHC16e2_sim dataset in 
`~/nitty_datasets.yml` which are written here:

 LHC16e2_sim:
   data_pattern: p20/AOD/*/AliAOD.Muons.root
   datadir: /alice/sim/2016/LHC16e2_plus/
   datatype: aod
   is_mc: 'true'
   notes: 5.02 TeV, Good runs with Muon quality. This is embedding
 simulation data.
   run_list: 246994, 246991, 246989, ... , 244982, 244980, 244918
   run_number_prefix: ''
   system: PbPb

 I checked this dataset on lxplus with my CERN account. They are here:
` /alice/sim/2016/LHC16e2_plus/run_number_6digits/p20/AOD/*/AliAOD.Muons.root`

 But the nittygriddy is not able to find them while I launch the downloading command: 
`nitty datasets --download LHC16e2_sim 1`

 Then it shows the message:
`=> Trying to connect to Server [0] root://pcapiserv03.cern.ch:10000 as User chhuang`
`DEBUG:root] Number of files found matching search string 0`
 
I think I input the correct directory in which the AliAOD.Muons.root are saved. However, it still cannot find the dataset.. Could you help me to understand what was happening?

 I would appreciate your help with this situation.
 Best regards,
 Chun-Lu Huang

**Comment by cbourjau** [comment]
Hi,
as I said already in the mail, I don't have a grid certificate anymore so I cannot test this. But I am pretty sure that the issue is with you `data_pattern` and whats happening in the background.
`nitty` actually does not download the file you specify directly but rather tries to find the zip archive, download that, and then unpacks it in the appropriate directory. (see `utils.py` and there the `download_from_grid_archive` and `find_associated_archive_files` functions). The rational here is that sometimes we have issues like this where you might want to have additional files and defining them individually would be a pain. But you probably still need to specify some sort of `data_pattern` when running on the grid (I think?). Maybe there is a smarter way to go about this...

Bottom line, can you try to change the `data_pattern` to `p20/AOD/*/AliAOD.root`? This might fix the download if the Muon files are in the same archive. I never used the muon files, so I don't know if you have to specify them explicitly, or if they are picked up automatically if you select the AliAOD.root files. Can you try? Also, I won't be able to reply in the next few days as I'm on the road.

Hope that helps for the time being!

**Comment by chhuangAli** [comment]
Hi,

Thanks for your fast reply with clear explanation. Actually it solved my issue after I replaced `AliAOD.Muons.root` by `AliAOD.root` in `data_pattern`. The `nitty` downloaded and extracted the zip archive as you explained. Those extracted files includes `AliAOD.Muons.root` also, which I need. Then I can launch my task to analyze them locally without problem. Hence I don't have to define extra files in those functions in `utils.py`. (I looked at them for a while. And you are right. That would be painful to define. ) 

Thanks for your help! I really appreciate it. 

Best wishes
